backend/services/table_anomaly_service.py

```
@@ -0,0 +1,433 @@
+                    SELECT column_name, data_type, is_nullable
+                    FROM information_schema.columns 
+                    WHERE table_name = %s AND table_schema = 'public'
+                """, (table_name,))
+                columns = cursor.fetchall()
+                
+                if not columns:
+                    return 0
+                
+                # NOT NULL 컬럼들에 대해 NULL 값 검사
+                null_count = 0
+                for column_name, data_type, is_nullable in columns:
```

Ruff (0.13.1)

Loop control variable `data_type` not used within loop body

Rename unused `data_type` to `_data_type`

(B007)